### PR TITLE
[MNG-6909] Use CDI to inject objects under tests

### PR DESCRIPTION
**Pull request by slachiewicz**
simplify tests - use Guice CDI to inject beans - instead of manually in setup methods

---
This change is [Reviewable](https://reviewable.io/reviews/apache/maven/346)

### CODE REVIEW

File: maven-core/src/test/java/org/apache/maven/project/PomConstructionTest.java [hunk 1]
```diff
@@ -53,12 +57,24 @@
...
     private RepositorySystem repositorySystem;
 
     private File testDirectory;
 
+    @Override
+    protected synchronized void setupContainer()
+    {
+        super.setupContainer();
+
+        ((DefaultPlexusContainer)getContainer())
```
Comment (elharo) [inline maven-core/src/test/java/org/apache/maven/project/PomConstructionTest.java:73]: This requires Java 8, and would probably be clearer without lambdas

Comment (slachiewicz) [reply maven-core/src/test/java/org/apache/maven/project/PomConstructionTest.java:73]: For Maven core we already have Java 8.. 
check here version without lambda: 

https://github.com/apache/maven/blob/master/maven-embedder/src/test/java/org/apache/maven/cli/MavenCliTest.java#L325-L335

**Comment by michael-o** [comment]
Why wasn't this merged yet?

**Comment by jira-importer** [comment]
Resolve #8622

**Pull request closed without merging**